src/array/cpu/spmm.cc

```diff
@@ -40,6 +40,60 @@ void SpMMCsr(const std::string& op, const std::string& reduce,
   }
 }
 
+/*! \brief Generalized SpMM on Csr format. */
+template <int XPU, typename IdType, int bits>
+void SpMMCsrHetero(const std::string& op, const std::string& reduce,
+             const BcastOff& bcast,
+             const std::vector<CSRMatrix>& vec_csr,
+             std::vector<NDArray> vec_ufeat,
+             std::vector<NDArray> vec_efeat,
+             std::vector<NDArray> vec_out,
+             std::vector<NDArray> out_aux,
+             const std::vector<dgl_type_t> ufeat_eid,
+             const std::vector<dgl_type_t> out_eid) {
```

Comment:
the names of the two arguments is kind of confusing. it seems they are eid, but they are actual node type ID of the source nodes and destination nodes, but the length of the vector is the number of edge type.

Reply:
can we just use something like `ufeat_node_tids` and `out_node_tids`?

Reply:
Done!

Reply:
i don't see the names changed.